Orange/widgets/utils/tests/test_filedialogs.py

```diff
@@ -0,0 +1,16 @@
+import os
+import unittest
+
+from Orange.widgets.utils.filedialogs import RecentPath
+
+
+class TestRecentPath(unittest.TestCase):
+    def test_resolve(self):
+        f = open("test.tab", 'w')
+        f.close()
+        recent_path = RecentPath(
+            os.path.join("temp/datasets", "test.tab"), "", "datasets/test.tab"
+        )
+        search_paths = [("basedir", os.getcwd())]
+        self.assertIsNotNone(recent_path.resolve(search_paths))
+        os.remove("test.tab")
```

Comment:
This looks like a classic trap of removing a file at the end, which might not happen if something goes wrong before the remove statement.
And here it is actually not that unusual since you have an assert above that might fail - try running this test on master.

I think there should be some utility functions for testing on temporary files somewhere... In any case some form of `finally: cleanup` is needed and I suggest using python's TemporaryFiles too.